src/archive_open.c

```diff
@@ -34,7 +38,7 @@ static int framed_stdin_open(struct archive *a, void *client_data)
 {
     struct fwup_archive_data *ad = (struct fwup_archive_data *) client_data;
 
-#if defined(__CYGWIN__) || defined(_WIN32)
+#ifdef __MINGW32__
```

Comment:
I think `_WIN32` also worked, but I was thinking it made sense to be consistent with the other `#ifdef`s I needed to add in other places. You probably know better than me whether it makes a difference in some cases. We certainly wouldn't need `__CYGWIN__` anymore if we're building on Linux instead of Cygwin.